client-runtime/src/main/java/com/microsoft/rest/v2/http/RxNettyClient.java

```diff
@@ -64,10 +73,21 @@ private SSLEngine getSSLEngine(String host) {
             rxnHeaders.put("Content-Length", Collections.<Object>singleton(String.valueOf(body.length())));
         }
 
+        // Is caching of rxnetty's HttpClient preferred for performance?
         boolean isSecure = "https".equalsIgnoreCase(uri.getScheme());
         io.reactivex.netty.protocol.http.client.HttpClient<ByteBuf, ByteBuf> rxnClient =
                 io.reactivex.netty.protocol.http.client.HttpClient.newClient(uri.getHost(), isSecure ? 443 : 80);
 
+        for (int i = 0; i < handlerConfigs.length; i++) {
+            ChannelHandlerConfig config = handlerConfigs[i];
+            if (config.mayBlock()) {
+                EventExecutorGroup executorGroup = new DefaultEventExecutorGroup(1);
```

Comment:
minor: Instead of creating one EventLoopGroup per handlers those belongs to `may-block` set, we could create one `DefaultEventExecutorGroup` with a fixed number of EventLoops (e.g. 10) and share it, that way threads will be pooled for `may-block` handlers.

Reply:
If desired, down the line we will be able to keep an EventExecutorGroup on the RxNettyClientAdapter instance with minimal effects outside the class.